static/js/typeahead_helper.js

```diff
@@ -207,17 +207,37 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+function activity_score(stream) {
+    var stream_score = 0;
+    if (stream.pin_to_top) {
+        stream_score += 2;
```

Comment:
The idea with the `+=` was that it could possibly return 3 to prioritize active pinned streams over inactive pinned streams. But if this extra level of sorting isn't useful, I can go ahead and remove it.